lib/config/options/index.ts

```diff
@@ -41,8 +41,16 @@ const options: RenovateOptions[] = [
     type: 'string',
     default: null,
     globalOnly: true,
-    allowedValues: ['alpha', 'created', 'updated', 'size', 'id'],
-    supportedPlatforms: ['forgejo', 'gitea'],
+    allowedValues: [
```

Comment:
this needs to be updated to reflect which values are allowed on which platform
https://github.com/renovatebot/renovate/blob/e272cea10f87668b6bd8067ce3322ea80176f512/docs/usage/self-hosted-configuration.md#L213

Reply:
created https://github.com/renovatebot/renovate/discussions/39709

Reply:
You are free to do the internal-logic refactoring in another PR. But, in the docs section of `autoDiscoverRepoSort` should be updated in this PR to specify which strings are available for which platforms. Like this:

gitlab: updated_at, created_at
forgejo & gitea: rest of the options ...